### PR TITLE
added sql server plugin audit storage authentication type

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/azure/sqlserver/auditStorageAuthType.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Auditing Storage Authentication Type',
+    category: 'SQL Server',
+    domain: 'Databases',
+    description: 'Ensures that managed identity is configured as authentication type when storage account is chosen as the destination for audit logs on SQL server.',
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.js:8]:
```suggestion
    description: 'Ensures that managed identity is configured as authentication type for storage account audit logs.',
```

File: plugins/azure/sqlserver/auditStorageAuthType.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    more_info: 'Enabling managed identity as authentication type enhances security when using a storage account as the destination for audit logs. Managed Identity can be a system-assigned managed identity or user-assigned managed identity.',
+    recommended_action: 'Ensure managed identity is configured as authentication type when choosing a storage account as the destination for audit logs on SQL server.',
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.js:10]:
```suggestion
    recommended_action: 'Ensure managed identity is configured as authentication type when choosing a storage account as the destination for audit logs on SQL server.',
```
maybe we should check for managed identity as authentication type for audit logs regardless of the destination

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/sqlserver/auditStorageAuthType.js (hunk 2 above)
Comment (muzzamilinovaqo) [reply plugins/azure/sqlserver/auditStorageAuthType.js:10]: storage account is the only destination type where managed identity or storage access keys can be configured. any other destination doesn't require any authentication type so.

File: plugins/azure/sqlserver/auditStorageAuthType.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        'Unable to query Auditing Policies: ' + helpers.addError(serverBlobAuditingPolicies), location, server.id);
+                } else {
+                    if (!serverBlobAuditingPolicies.data.length) {
+                        helpers.addResult(results, 0, 'No Server Auditing policies found', location, server.id);
+                    } else {
+                        let serverAuditingEnabled = serverBlobAuditingPolicies.data.length && serverBlobAuditingPolicies.data.find(auditPolicy => auditPolicy.state && auditPolicy.state.toLowerCase() == 'enabled');
+                        if (serverAuditingEnabled) {
+                            if (serverAuditingEnabled.storageAccountSubscriptionId !== '00000000-0000-0000-0000-000000000000') {
+                                if (serverAuditingEnabled.isManagedIdentityInUse) {
+                                    helpers.addResult(results, 0, 'Managed identity is configured as authentication type for audit logs storage on SQL server', location, server.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.js:53]:
```suggestion
                                    helpers.addResult(results, 0, 'SQL Server is using managed identity authentication for storage account audit logs', location, server.id);
```

File: plugins/azure/sqlserver/auditStorageAuthType.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                                } else {
+                                    helpers.addResult(results, 2, 'Managed identity is not configured as authentication type for audit logs storage on SQL server', location, server.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.js:55]:
```suggestion
                                    helpers.addResult(results, 2, 'SQL Server is not using managed identity authentication for storage account audit logs', location, server.id);
```

File: plugins/azure/sqlserver/auditStorageAuthType.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                                }
+                            } else {
+                                helpers.addResult(results, 0, 'Auditing is not using storage account for SQL server', location, server.id);
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.js:58]:
```suggestion
                                helpers.addResult(results, 0, 'SQL Server is not using a storage account as destination for audit logs', location, server.id);
```

File: plugins/azure/sqlserver/auditStorageAuthType.spec.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+            );
+
+            auditLogsAuthentication.run(cache, {}, callback);
+        });
+
+        it('should give failing result if managed identity is not configured for audit logs storage', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Managed identity is not configured as authentication type for audit logs storage on SQL server');
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.spec.js:139]:
```suggestion
                expect(results[0].message).to.include('SQL Server is not using managed identity authentication for storage account audit logs');
```

File: plugins/azure/sqlserver/auditStorageAuthType.spec.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+            );
+
+            auditLogsAuthentication.run(cache, {}, callback);
+        });
+
+        it('should give passing result if managed identity is configured for audit logs storage', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Managed identity is configured as authentication type for audit logs storage on SQL server');
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.spec.js:93]:
```suggestion
                expect(results[0].message).to.include('SQL Server is using managed identity authentication for storage account audit logs');
```

File: plugins/azure/sqlserver/auditStorageAuthType.spec.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+            );
+
+            auditLogsAuthentication.run(cache, {}, callback);
+        });
+
+        it('should give passing result if Azure SQL Auditing is not using account storage', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Auditing is not using storage account for SQL server');
```
Comment (mehakseedat63) [inline plugins/azure/sqlserver/auditStorageAuthType.spec.js:47]:
```suggestion
                expect(results[0].message).to.include('SQL Server is not using a storage account as destination for audit logs');
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**